From: Juergen Beisert <jbe@pengutronix.de>
Subject: interim solution to fix a broken 'round closest'

Since commit 263a523d18bca306016d75f5c8d5c57c37fe52fb DIV_ROUND_CLOSEST()
returns 4198403 when the ADC delivers a '0'. Don't know the correct fix yet,
so, handle the '0' in a special way.

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>

---
 drivers/hwmon/s3c-hwmon.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

Index: linux-3.6/drivers/hwmon/s3c-hwmon.c
===================================================================
--- linux-3.6.orig/drivers/hwmon/s3c-hwmon.c
+++ linux-3.6/drivers/hwmon/s3c-hwmon.c
@@ -179,8 +179,10 @@ static ssize_t s3c_hwmon_ch_show(struct 
 	if (ret < 0)
 		return ret;
 
-	ret *= cfg->mult;
-	ret = DIV_ROUND_CLOSEST(ret, cfg->div);
+	if (ret != 0) {
+		ret *= cfg->mult;
+		ret = DIV_ROUND_CLOSEST(ret, cfg->div);
+	}
 
 	return snprintf(buf, PAGE_SIZE, "%d\n", ret);
 }
